shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/Server.java

```diff
@@ -2631,11 +2633,28 @@ public void downloadInputData(
       try (var os = t.getResponseBody()) {}
       return;
     }
-    t.getResponseHeaders().set("Content-type", "application/json");
-    t.sendResponseHeaders(200, 0);
+
+    // If the format opts to use ErrorableStream itself, OK may be true or false. Otherwise,
+    // constructor will automatically set OK to true
+    ErrorableStream<Object> fetchedInput =
+        new ErrorableStream<>(inputSource.fetch(format.name(), readStale));
+    if (fetchedInput.isOk()) {
+      t.getResponseHeaders().set("Content-type", "application/json");
+      t.sendResponseHeaders(200, 0);
+    } else {
+      t.sendResponseHeaders(503, 0);
+      try (var os = t.getResponseBody()) {}
+      /**
+       * FIXME Previously, writeJson included a call to fetch, now we download the input much
+       * earlier. Do i need to release the inputDownloadSemaphore for any failures here too? I think
+       * this increments the semaphore too much but I can't tell what the proper behaviour is
+       */
+      inputDownloadSemaphore.release();
```

Comment:
* `os` is unused
* semaphore acquisition and release looks like it should be fine to me. If a permit isn't acquired, the method returns early. If a permit is acquired, it should be released in one of two places. I don't see how it could be released twice

Reply:
Looks like getting the response body closes the stream (https://docs.oracle.com/en/java/javase/17/docs/api/jdk.httpserver/com/sun/net/httpserver/HttpExchange.html#getResponseBody()) and in its original implementation it nicely echoed the format of the else block: https://github.com/oicr-gsi/shesmu/pull/519/files#diff-8800ecd9470735edffb75cb231ef63395c02e230a1e4070f5fc9a0e90217c3d8

Could see about calling close() instead, but it looks like this is a valid way to close the HttpExchange too

Reply:
Ah, ok. Getting the response body doesn't close the stream, but it's in a try-with-resources statement, so it gets auto-closed after that (empty) block completes. Calling `close` would have a more obvious purpose